### PR TITLE
Gravpults and some Centcomm detailling

### PR DESCRIPTION
**Pull request by DeityLink**
image

dreamseeker 2018-07-01 03-32-37-51

image

image

image

Worth mentioning that this PR opens the way for new contextual HUD buttons like the Gravpult's "Launch"

:cl:
* rscadd: The deathsquad's gravpults are back, and working better than they ever did! Walk onto them, click on the holomap to pinpoint your destination, then press Launch!
* rscadd: The deathsquad now has its own armory, with a few rare/high tier weapons, should they ever want something to compliment their pulse rifle.
* tweak: Tweaked Centcomm's recreational area a bit.
* bugfix: Fixed an issue that could cause rocket dashing mechs to get thrown backwards back into space.

### CODE REVIEW

File: code/_onclick/hud/screen_objects.dm [hunk 1]
```diff
@@ -60,6 +60,42 @@
...
+
+
+
+///////////////////////////////////////
+/obj/abstract/screen/interface
+	var/mob/user = null
+	var/obj/machine = null
+	var/action = ""
+
+/obj/abstract/screen/interface/New(turf/loc,u,m,a,i,i_s="",l="CENTER,CENTER",px=0,py=0)
```
Comment (MadmanMartian) [inline code/_onclick/hud/screen_objects.dm:99]: Okay, do a doc-comment above this proc to say what each of these variables mean, as single-letter variables in a function's gonna be a bitch to maintain by anyone else.

**Review by MadmanMartian — COMMENTED** [review]
Nothing glaring code-wise from what I could see.

File: code/_onclick/hud/screen_objects.dm (hunk 1 above)
Comment (DeityLink) [reply code/_onclick/hud/screen_objects.dm:99]: Done https://github.com/DeityLink/vgstation13-1/blob/13cfb97ef36fd88b1ae7da6e8ed1de1ac95fd576/code/_onclick/hud/screen_objects.dm#L75

**Comment by Kurfursten** [comment]
I don't like making the blast doors to the rest of centcomm transparent

I use those opaque doors for appearing when I need to give a briefing as Commander Green

**Comment by DeityLink** [comment]
Why couldn't you still do it? Plus you can always replace the door manually if you're gonna do some bus. But by default I like reminding strike teams that they're on centcomm.

**Comment by Kurfursten** [comment]
Now it is much more obvious when I teleport over the body and resurrect it, since they can see through the doors before they open

**Comment by DeityLink** [comment]
For the whole 0.2sec it takes to press the button to open the door? :/

**Comment by DeityLink** [comment]
OH and before I forget, deathsquads wear thermals so they can see you coming anyway....

**Comment by Kurfursten** [comment]
You changed the ERT side too though, and yes.

Why's it so important the blast door be transparent? So lasers can be fired into the centcomm halls?

**Comment by DeityLink** [comment]
They can't. Only the opacity is changed. Like I said if you want to bus, you can spawn another blast door and give it the same id_tag.
It's an aesthetics choice. I was the one who placed these doors and corridors in the first place for the aforementioned reason, but just like you don't like being able to see behind I don't like not being able to see behind.

**Comment by Tokanova** [comment]
if i launch a mech with the mass driver in the chapel and it hits the derelict, will it smash through the walls?

**Comment by DeityLink** [comment]
Windows, grilles and tables only.

And it's been able to do that for over a year. See #15491

**Comment by SonixApache** [comment]
awesome

**Comment by Duny-** [comment]
I never thought I would see these things actually work in my lifetime, and you didn't even half-ass it
Truly this is witchcraft

**Comment by MusgardTiger** [comment]
blessed PR

**Comment by SonixApache** [comment]
wheeeen

**Comment by jknpj** [comment]
now

**Comment by DeityLink** [comment]
Yessss, can't wait to be home tonight.

**Pull request merged**